mlflow/pyfunc/__init__.py

```diff
@@ -320,62 +433,192 @@ def predict(*args):
+    :ref:`"which workflow is right for my use case?" <pyfunc-create-custom-selecting-workflow>`.
+    Note that the parameters for the first workflow: ``loader_module``, ``data_path`` and the
+    parameters for the second workflow: ``python_model``, ``artifacts``, cannot be
+    specified together.
+
+    :param dst_path: The path to which to save the Python model.
+    :param loader_module: The name of the Python module that will be used to load the model
+                          from ``data_path``. This module must define a method with the prototype
+                          ``_load_pyfunc(data_path)``. If not *None*, this module and its
+                          dependencies must be included in one of the following locations:
+
+                          - The MLflow library.
+                          - Package(s) listed in the model's Conda environment, specified by
+                            the ``conda_env`` parameter.
+                          - One or more of the files specified by the ``code_path`` parameter.
 
-    :param dst_path: Path where the model is stored.
-    :param loader_module: The module to be used to load the model.
     :param data_path: Path to a file or directory containing model data.
-    :param code_path: List of paths (file or dir) contains code dependencies not present in
-                      the environment. Every path in the ``code_path`` is added to the Python
+    :param code_path: A list of local filesystem paths to Python file dependencies (or directories
+                      containing file dependencies). These files will be *prepended* to the system
                       path before the model is loaded.
-    :param conda_env: Path to the Conda environment definition. This environment is activated
-                      prior to running model code.
-    :return: Model configuration containing model info.
-
+    :param conda_env: Either a dictionary representation of a Conda environment or the path to a
+                      Conda environment yaml file. If provided, this decribes the environment
+                      this model should be run in. At minimum, it should specify the dependencies
+                      contained in :data:`mlflow.pyfunc.DEFAULT_CONDA_ENV`. If `None`, the default
+                      :data:`mlflow.pyfunc.DEFAULT_CONDA_ENV` environment will be added to the
+                      model. The following is an *example* dictionary representation of a Conda
+                      environment::
+
+                        {
+                            'name': 'mlflow-env',
+                            'channels': ['defaults'],
+                            'dependencies': [
+                                'python=3.7.0',
+                                'cloudpickle==0.5.8'
+                            ]
+                        }
+
+    :param python_model: An instance of a subclass of :class:`~PythonModel`. This class will be
```

Comment:
for some reason this looks weird (py3 make):

![image](https://user-images.githubusercontent.com/1400247/51126064-0ae99b00-17d7-11e9-8068-5988b123b721.png)